pkg/provider/github/acl_test.go

```diff
@@ -261,3 +262,96 @@ func TestAclCheckAll(t *testing.T) {
 		})
 	}
 }
+
+func TestIfPullRequestIsForSameRepoWithoutFork(t *testing.T) {
+	tests := []struct {
+		name              string
+		event             *info.Event
+		commitFiles       []*github.CommitFile
+		pullRequest       *github.PullRequest
+		pullRequestNumber int
+		allowed           bool
+		wantError         bool
+	}{
+		{
+			name: "when pull request raised by non owner to the repository where non owner did not fork but have permission to create branch",
+			event: &info.Event{
+				Organization:      "owner",
+				Sender:            "nonowner",
+				Repository:        "pullrequestrepository",
+				PullRequestNumber: 1,
+			},
+			pullRequest: &github.PullRequest{
+				ID:     github.Int64(1234),
+				Number: github.Int(1),
+				Head: &github.PullRequestBranch{
+					Ref: github.String("main"),
+					Repo: &github.Repository{
+						CloneURL: github.String("http://org.com/owner/repo"),
+					},
+				},
+				Base: &github.PullRequestBranch{
+					Ref: github.String("dependabot"),
+					Repo: &github.Repository{
+						CloneURL: github.String("http://org.com/owner/repo"),
+					},
+				},
+			},
+			pullRequestNumber: 1,
+			allowed:           true,
+			wantError:         false,
+		}, {
+			name: "failed to get Pull Request",
+			event: &info.Event{
+				Organization:      "owner",
+				Sender:            "nonowner",
+				Repository:        "pullrequestrepository",
+				PullRequestNumber: 2,
+			},
+			pullRequest: &github.PullRequest{
+				ID:     github.Int64(1234),
+				Number: github.Int(1),
+				Head: &github.PullRequestBranch{
+					Ref: github.String("main"),
+					Repo: &github.Repository{
+						CloneURL: github.String("http://org.com/owner/repo"),
+					},
+				},
+				Base: &github.PullRequestBranch{
+					Ref: github.String("dependabot"),
+					Repo: &github.Repository{
+						CloneURL: github.String("http://org.com/owner/repo"),
+					},
+				},
+			},
```

Comment:
maybe check where user is not allowed? I think it test only for allowance and if repository is notfound?